src/crypto/mbedtls/verifier.cpp

```diff
@@ -105,4 +106,64 @@ namespace crypto
 
     return Pem(buf, len);
   }
-}
+
+  bool Verifier_mbedTLS::verify_certificate(
+    const std::vector<const Pem*>& trusted_certs)
+  {
+    int rc = 0;
+
+    mbedtls_x509_crt* chain = NULL;
+
+    if (!trusted_certs.empty())
+    {
+      chain = new mbedtls_x509_crt[trusted_certs.size()];
+
+      for (size_t i = 0; i < trusted_certs.size(); i++)
+      {
+        mbedtls_x509_crt_init(&chain[i]);
+      }
+
+      for (size_t i = 0; i < trusted_certs.size(); i++)
+      {
+        auto& tc = trusted_certs[i];
+        if (
+          (rc = mbedtls_x509_crt_parse(&chain[i], tc->data(), tc->size())) != 0)
+        {
+          goto exit;
```

Comment:
Could we avoid the C-style `goto` and `exit` label and use the mbedtls wrappers if possible?

Reply:
Not easy; mbedTLS expects a buffer full of it's own data structures, it wouldn't work if we need more space per object in the buffer. In theory we could add a new wrapper for this type of set, but this is the only place we need it.